src/conn/conn_ckpt.c

```diff
@@ -78,14 +94,20 @@ __ckpt_server(void *arg)
 	conn = S2C(session);
 	wt_session = (WT_SESSION *)session;
 
-	while (F_ISSET(conn, WT_CONN_SERVER_RUN) &&
-	    F_ISSET(conn, WT_CONN_SERVER_CHECKPOINT)) {
+	for (;;) {
+		if (!__ckpt_server_run_chk(session))
+			break;
```

Comment:
It could be, but the loop begins like this:
```
for (;;) {
        if (!__ckpt_server_run_chk(session))
                break;
        /*
         * Wait...
         * NOTE: If the user only configured logsize, then usecs
         * will be 0 and this wait won't return until signalled.
         */
        __wt_cond_wait(session,
            conn->ckpt_cond, conn->ckpt_usecs, __ckpt_server_run_chk);

        /* Check if we're quitting or being reconfigured. */
        if (!__ckpt_server_run_chk(session))
                break;
```
That is, `__ckpt_server_run_chk()` appears 3 times in the first few lines of the loop, and I hoped repeating the construct would make things clearer, than calling the function different ways.